api/kv_v2.go

```diff
@@ -238,6 +262,48 @@ func (kv *kvv2) Put(ctx context.Context, secretPath string, data map[string]inte
 	return kvSecret, nil
 }
 
+// PutMetadata can be used to fully replace a subset of metadata fields for a
+// given KV v2 secret. All fields will replace the corresponding values on the Vault server.
+// Any fields left as nil will reset the field on the Vault server back to its zero value.
+//
+// To only partially replace the values of these metadata fields, use PatchMetadata.
+//
+// This method can also be used to create a new secret with just metadata and no secret data yet.
+func (kv *kvv2) PutMetadata(ctx context.Context, secretPath string, metadata KVMetadataInput) error {
+	pathToWriteTo := fmt.Sprintf("%s/metadata/%s", kv.mountPath, secretPath)
+
+	// convert values to a map we can pass to Logical,
+	// casting nils to the appropriate zero value
+	md := make(map[string]interface{})
```

Comment:
Nit: since the variable is used throughout the scope of the function, I think a longer name is warranted. `metadataMap` maybe? One or two letter names are usually used for short for loop blocks.